### DISCUSSION TITLE
LVGL + micropython + ESP32 - RAM error while compiling big files

### DISCUSSION THREAD

**Discussion by glatocha** [micropython/micropython#11277] [Libraries & Drivers]
Hello all,
I have a board with 2.8" ILI9341 screen with touch and the ESP-WROOM-32. The ESP module has no additional RAM.
I have successfully compiled micropython, downloaded. Wrote basic code with ILI9341 driver and xtp touch. Everything is ok.
So I am using squareLine studio to generate screen. One screen is ok with touch, all good. Now for my application I prepared 5 screens, the export is about 1000 lines of code. During "import ui.py" I got the error of exceeding memory size. 

So here is my question:
What is the proper technique of splitting and fragmenting the code to multiple files to get the compilation on import done. 
Right now I have it split to basically 3 files with extra one to offset all ui functions. The Screens objects and screen changing buttons I have put in the global file to be able to switch screens. This obviously gives a lot work on each squareLine export. And I am not sure if it proper technique anyway. I do not want to precompile it as I will need to work on it.

**Comment by peterhinch** [comment]
Try the [cross-compiler](http://docs.micropython.org/en/latest/reference/mpyfiles.html). You compile on the PC and put the resultant .mpy file on the target. This reduces the RAM usage because you aren't running the compiler on the target.
